packages/devtools-reps/src/object-inspector/index.js

```diff
@@ -196,12 +209,29 @@ class ObjectInspector extends Component {
     if (expand === true) {
       const {
         getObjectProperties,
+        getObjectEntries,
         loadObjectProperties,
+        loadObjectEntries,
       } = this.props;
 
       const value = getValue(item);
+      const parent = getParent(item);
+      const parentValue = getValue(parent);
+      const parentActor = parentValue
+        ? parentValue.actor
+        : null;
+
       if (nodeHasProperties(item) && value && !getObjectProperties(value.actor)) {
         loadObjectProperties(value);
+        return;
+      }
+
+      if (
+        nodeIsEntries(item)
+        && !hasAllEntriesInPreview(parent)
+        && !getObjectEntries(parentActor)
+      ) {
+        loadObjectEntries(parentValue);
```

Comment:
I did it that way because it makes it clear what the consumer needs to load.
If we expect to load entries from `loadObjectProperties`, that means the consumer has to check what kind of data it needs to load.
Now that I think about it, we might be able to do this with a couple of changes.

We should change `getObjectProperties(actor:string)` to `getObjectProperties(path:string)`, which means the consumer needs to keep track of the paths it has already loaded.

In order to do that, we need to change `loadObjectProperties(value:Grip)` to `loadObjectProperties(value:Grip, path:string, nodeTypes: Array<nodeType>)`.
`nodeTypes` would be the inversed tree of the nodeTypes for the path we want to load properties.It would help us when/if we decide to lazy load bucketed items (e.g. _"Let's load the 100-199 slice of the `<entries>` node"_ could be done because `nodeTypes` would be something like `[bucket, entries, grip]`).
The `nodeType` would be pulled of a `type` property on the node (which can be one of bucket, entries, get, set, promiseState, promiseValue, promiseReason. …), which will be nice to have too since it will allow us to rely on it instead of the path to check if a node is a bucket, or an `entries` node for example.
We should then expose those types so the consumer can test against them in order to fetch the correct data.

What do you think @jasonLaster ?